test/e2e/specs/site-editor/font-library.spec.js

```diff
@@ -15,15 +15,64 @@ test.describe( 'Font Library', () => {
 		} );
 
 		test( 'should display the "Manage Fonts" icon', async ( { page } ) => {
-			await page.getByRole( 'button', { name: /styles/i } ).click();
+			await page.getByRole( 'button', { name: 'Styles' } ).click();
 			await page
-				.getByRole( 'button', { name: /typography styles/i } )
+				.getByRole( 'button', { name: 'Typography Styles' } )
 				.click();
 			const manageFontsIcon = page.getByRole( 'button', {
-				name: /manage fonts/i,
+				name: 'Manage Fonts',
 			} );
 			await expect( manageFontsIcon ).toBeVisible();
 		} );
+
+		test( 'should allow user to upload a local font', async ( {
+			page,
+		} ) => {
```

Comment:
Could we ensure the fonts directory is delete prior to running this test. This ensures correct coverage for https://github.com/WordPress/gutenberg/issues/59023

Reply:
We can create a e2e test Plugin that implements the testing instruction using a filter as provided here

https://github.com/WordPress/gutenberg/issues/59023#issuecomment-2018324697

Activate the plugin prior to test and deactivate afterwards.

FYI this is an established pattern even if it requires jumping through a few hoops. Luckily all Plugin activation happens via API rather than programmatic interaction so it's fast(er).